### PR TITLE
Tonyvanswet/ele 1508 carbon intensity legend

### PR DESCRIPTION
**Pull request by tonypls**
## Issue
https://linear.app/electricitymaps/issue/ELE-1508/carbon-intensity-legend
## Description
Duplicate of the old legend with some types

### CODE REVIEW

**Comment by VIKTORVAV99** [comment]
So eeh, we updated this in https://github.com/electricitymaps/electricitymaps-contrib/pull/4789 which was merged after your PR was created. It would be great if you could bring over those changes as well (they aren't very big so should be pretty straight forward).

File: web/src/components/legend/Legend.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import type { ReactElement } from 'react';
+import Co2Legend from './Co2Legend';
+
+export default function Legend(): ReactElement {
```
Comment (madsnedergaard) [inline web/src/components/legend/Legend.tsx:4]: Can we call it something like this so it is clear that it's a "holder of legends"? :)
```suggestion
export default function LegendsContainer(): ReactElement {
```

File: web/src/main.tsx [hunk 2]
```diff
@@ -48,7 +48,7 @@ if (container) {
             <App />
           </AtomsDevtools>
         </BrowserRouter>
-        <ReactQueryDevtools position="bottom-right" initialIsOpen={false} />
+        <ReactQueryDevtools position="top-right" initialIsOpen={false} />
```
Comment (madsnedergaard) [inline web/src/main.tsx:51]: I love how this thing keeps being pushed around 😂

File: web/src/components/legend/Co2Legend.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      {children}
+    </div>
+  );
+}
+
+export default function Co2Legend(): ReactElement {
+  const { __ } = useTranslation();
+  const co2ColorScale = useCo2ColorScale();
+  return (
+    <div className="flex h-[78px] w-[224px] rounded bg-white p-2 shadow dark:bg-gray-900">
```
Comment (madsnedergaard) [inline web/src/components/legend/Co2Legend.tsx:31]: Should this styling be on the "Legends Holder" component instead?

Comment (tonypls) [reply web/src/components/legend/Co2Legend.tsx:31]: Ah yeah you're right, I thought it made sense to have transparent spacing between different legends but after checking that's not how we currently do it.

Comment (tonypls) [reply web/src/components/legend/Co2Legend.tsx]: updated!

File: web/src/components/legend/Legend.tsx (hunk 1 above)
Comment (tonypls) [reply web/src/components/legend/Legend.tsx]: added

File: web/src/main.tsx (hunk 2 above)
Comment (tonypls) [reply web/src/main.tsx:59]: maybe we should set a randomizer

**Review by madsnedergaard — APPROVED** [review]

**Pull request merged**